### ISSUE TITLE
MissingParamType shouldn't be reported when implementing an interface or overriding a parent class method

### ISSUE THREAD

**Issue by ndench** [vimeo/psalm#509]
I have a class implementing the `\Serializable` interface:

```php
class User implements \Serializable
{
    private $id;
    private $email;
    private $password;

    public function unserialize($serialized)
    {  
        [
            $this->id,
            $this->email,
            $this->password,
        ] = \unserialize((string) $serialized);
    }
}
```

However I get this error reported:
> MissingParamType: Parameter $serialized has not provided type

The function definition must be compatible with the one defined in the `\Serializable` interface, so this cannot be fixed.

**Comment by muglug** [comment]
You can always add `/** @param mixed $serialized */`, but I'll fix this.
